### DISCUSSION TITLE
Real world use cases of `test.fails` usage

### DISCUSSION THREAD

**Discussion by vasilii-kovalev** [vitest-dev/vitest#1427] [Q&A]
I'd like to know what real world use cases for [test.fails](https://vitest.dev/api/#test-fails) function can be.
If it is necessary to skip some tests because there is no time to fix them, [test.skip](https://vitest.dev/api/#test-skip) can be used. It even shows, that there are tests to fix, in the console, while `test.fails` hides such technical debt...

**Comment by dermoumi** [comment]
It's grammar sugar for sentences like: `it.fails("to render correctly", () => { ... });`

**Reply by meduzen** [reply]
Came here because I wondered the same thing. My use case is similar:

- I spotted an issue in a library I maintain.
- Awaiting to code a fix, I already want to add tests in my test suite for the failing use case.
- I want Vitest to know that these tests currently fails, so it’s “normal for now” and the CI should not flag these tests as a blocking point.

So I thought using `test.fails` would do the trick, but it actually makes the CI crash:

<details>
<summary>CI output</summary>

```sh
> vitest run


 RUN  v3.0.7 /home/runner/work/frontacles/frontacles

 ✓ src/math/math.test.js (26 tests) 7ms
 ❯ src/url/email.test.js (16 tests | 1 failed) 8ms
   ✓ url/email > Email class
   ✓ url/email > email username
   ✓ url/email > email hostname
   ✓ url/email > destructure { username, hostname}
   ✓ url/email > string using type casting
   ✓ url/email > string using `.toString`
   ✓ url/email > passes the Zod test suits for valid emails
   ✓ url/email > passes the Zod test suits for invalid emails
   ✓ url/email > can parse o&leary@example.com
   ✓ url/email > can parse an Email object
   ✓ url/email > can’t parse a Date
   ✓ url/email > can’t parse a URL string with a port
   ✓ url/email > can’t parse a URL string with no username and with a port
   ✓ url/email > can’t parse a URL string with password
   × url/email > to parse without username 2ms
     → Expect test to fail
   ✓ url/email > username and domain are mutable

⎯⎯⎯⎯⎯⎯⎯ Failed Tests 1 ⎯⎯⎯⎯⎯⎯⎯

 FAIL  src/url/email.test.js > url/email > to parse without username
Error: Expect test to fail
⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[1/1]⎯


 Test Files  1 failed | 1 passed (2)
      Tests  1 failed | 41 passed (42)
   Start at  10:30:02
   Duration  286ms (transform 53ms, setup 0ms, collect 70ms, tests 15ms, environment 0ms, prepare 160ms)

Error: Process completed with exit code 1.
```

</details>

So I’m a bit confused, too, as what purpose it covers. I switched to `test.skip`.
